src/components/Resource/ResourceFilter.tsx

```diff
@@ -16,7 +16,7 @@ import { RESOURCE_FILTER_ORDER } from "@/common/constants";
 import { RESOURCE_CHOICES } from "@/common/constants";
 
 import routes from "@/Utils/request/api";
-import useTanStackQueryInstead from "@/Utils/request/useQuery";
+import useQuery from "@/Utils/request/useQuery";
```

Comment:
this is just renaming the import. use useQuery from tanstack query instead.

https://docs.ohc.network/docs/care/development/data-fetching-in-care/

Reply:
@rithviknishad  did we implementated this anywhere in care_fe for reference ?

In the current version of tanstack the `onSuccess`, `onError`, and `onSettled `callbacks have been removed from the useQuery options. I got to know that these callbacks can now be handled using `useEffect` or other hooks.

is below approach acceptable?

```js
  const { isLoading: orginFacilityLoading, data } = useQuery({
    queryKey: [routes.getAnyFacility.path, filter.origin_facility],
    queryFn: () =>
      query(routes.getAnyFacility, {
        pathParams: { id: filter.origin_facility },
      }),
    enabled: filter.origin_facility !== undefined,
  });

  useEffect(() => {
    if (data) {
      setFilterState({
        origin_facility_ref: filter.origin_facility === "" ? "" : data,
      });
    }
  }, [data]);

Reply:
Yea, that approach works.